File: src/main/java/com/ggardet/patient/internal/entity/Medication.java
package com.ggardet.patient.internal.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.util.UUID;

@Entity
@Data
@Table(name = "medication", schema = "patients")
public class Medication {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @Column(name = "patient_id")
    private UUID patientId;
    @Column(nullable = false)
    private String name;
    @Column
    private String dosage;
    @Column
    private String frequency;
    @Column(name = "start_date")
    private LocalDate startDate;
    @Column(name = "end_date")
    private LocalDate endDate;
    @Column(name = "prescribing_doctor_id")
    private UUID prescribingDoctorId;
    @Column
    private String instructions;
}

File: src/main/kotlin/com/ggardet/patient/internal/entity/Medication.kt
package com.ggardet.patient.internal.entity

import jakarta.persistence.*
import java.time.LocalDate
import java.util.*

@Entity
@Table(name = "medication", schema = "patients")
data class Medication(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null,
    
    @Column(name = "patient_id")
    var patientId: UUID? = null,
    
    @Column(nullable = false)
    var name: String,
    
    @Column
    var dosage: String? = null,
    
    @Column
    var frequency: String? = null,
    
    @Column(name = "start_date")
    var startDate: LocalDate? = null,
    
    @Column(name = "end_date")
    var endDate: LocalDate? = null,
    
    @Column(name = "prescribing_doctor_id")
    var prescribingDoctorId: UUID? = null,
    
    @Column
    var instructions: String? = null
) {
    // JPA requires a no-arg constructor
    constructor() : this(name = "")
}